gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/resources/ResourcesSpec.kt

```diff
@@ -1,44 +1,95 @@
 package org.jetbrains.compose.resources
 
-import com.squareup.kotlinpoet.ClassName
-import com.squareup.kotlinpoet.CodeBlock
-import com.squareup.kotlinpoet.FileSpec
-import com.squareup.kotlinpoet.KModifier
-import com.squareup.kotlinpoet.PropertySpec
-import com.squareup.kotlinpoet.TypeSpec
-import com.squareup.kotlinpoet.withIndent
+import com.squareup.kotlinpoet.*
 import java.nio.file.Path
 import kotlin.io.path.invariantSeparatorsPathString
-import kotlin.io.path.pathString
 
 internal enum class ResourceType(val typeName: String) {
-    IMAGE("images"),
-    STRING("strings"),
-    FONT("fonts");
+    DRAWABLE("drawable"),
+    STRING("string"),
+    FONT("font");
+
+    override fun toString(): String = typeName
 
     companion object {
-        fun fromString(str: String) = when (str) {
-            "images" -> ResourceType.IMAGE
-            "strings" -> ResourceType.STRING
-            "fonts" -> ResourceType.FONT
-            else -> error("Unknown resource type: $str")
-        }
+        fun fromString(str: String): ResourceType =
+            ResourceType.values()
+                .firstOrNull { it.typeName.equals(str, true) }
+                ?: error("Unknown resource type: $str")
     }
 }
 
 internal data class ResourceItem(
     val type: ResourceType,
-    val qualifiers: Set<String>,
+    val qualifiers: List<String>,
     val name: String,
     val path: Path
 )
 
 private fun ResourceItem.getClassName(): ClassName = when (type) {
-    ResourceType.IMAGE -> ClassName("org.jetbrains.compose.resources", "ImageResource")
+    ResourceType.DRAWABLE -> ClassName("org.jetbrains.compose.resources", "DrawableResource")
     ResourceType.STRING -> ClassName("org.jetbrains.compose.resources", "StringResource")
     ResourceType.FONT -> ClassName("org.jetbrains.compose.resources", "FontResource")
 }
 
+private fun CodeBlock.Builder.addQualifiers(resourceItem: ResourceItem): CodeBlock.Builder {
+    val languageQualifier = ClassName("org.jetbrains.compose.resources", "LanguageQualifier")
+    val regionQualifier = ClassName("org.jetbrains.compose.resources", "RegionQualifier")
+    val themeQualifier = ClassName("org.jetbrains.compose.resources", "ThemeQualifier")
+    val densityQualifier = ClassName("org.jetbrains.compose.resources", "DensityQualifier")
+
+    val languageRegex = Regex("[a-z][a-z]")
```

Comment:
As we are in JVM, we may use available locales in the JVM for locale qualifiers checking instead of rexexp

Reply:
I wouldn't trust JVM locales because it depends on a version OS and JVM. I understa nd in our case it is safe enough but the regex isn't complex here